openstack/orchestration/v1/resourcetypes/results.go

```diff
@@ -0,0 +1,163 @@
+package resourcetypes
+
+import (
+	"github.com/gophercloud/gophercloud"
+)
+
+// ResourceTypeList contains the result of listing the available resource types.
+// It includes both a list of resource types and a map of their descriptions.
+// The descriptions will be empty unless they were requested from the server.
+type ResourceTypeList struct {
+	ResourceTypes []string
+	Descriptions  map[string]string
+}
+
+// PropertyType represents the expected type of a property or attribute value.
+type PropertyType string
```

Comment:
It's not strictly necessary, I don't think. It seemed useful as documentation, so that callers can see the list of possible values and also to have constants already defined to use in switch statements, which is something that ~every user of this code would otherwise have to write for themselves.